### PR TITLE
Fix : SFTP Sensor fails to locate file when file_pattern is provided

### PR DESCRIPTION
**Pull request by ajinkya-xyz**
Background : Pull request # [24084](https://github.com/apache/airflow/pull/24084) added the ability to provide `fnamtch` file pattern to SFTP sensor. But sensor fails to locate the file when the file pattern is provided. 
 
Issue : Code fails to get modified time once file matching to the given pattern is found. [Line # 77](https://github.com/apache/airflow/blob/main/airflow/providers/sftp/sensors/sftp.py#L77) from sftp.py file fails with no such file error. 

`mod_time = self.hook.get_mod_time(actual_file_to_check)`

Root cause: Code assumes that the `get_file_by_pattern` method returns a complete path for a file matching the given `fnamtch` expression. While we are only getting file name in return. `get_file_by_pattern` internally relies on [Paramiko SFTP clients listdir()](https://docs.paramiko.org/en/stable/api/sftp.html#paramiko.sftp_client.SFTPClient.listdir) method to retrieve files contained in given path folder. `listdir()` method only returns file names and not the complete file path. 

Related code, sftp.py file [line # 68.](https://github.com/apache/airflow/blob/main/airflow/providers/sftp/sensors/sftp.py#L68)
`file_from_pattern = self.hook.get_file_by_pattern(self.path, self.file_pattern)`

Fix : Prepending the path to `actual_file_to_check `variable fixes the issue. 

`actual_file_to_check = os.path.join(self.path, file_from_pattern)`

Related : [24084](https://github.com/apache/airflow/pull/24084)

### CODE REVIEW

File: airflow/providers/sftp/sensors/sftp.py [hunk 1]
```diff
@@ -67,7 +68,7 @@ def poke(self, context: 'Context') -> bool:
         if self.file_pattern:
             file_from_pattern = self.hook.get_file_by_pattern(self.path, self.file_pattern)
             if file_from_pattern:
-                actual_file_to_check = file_from_pattern
+                actual_file_to_check = os.path.join(self.path, file_from_pattern)
```
Comment (uranusjr) [inline airflow/providers/sftp/sensors/sftp.py]: Hmm, is it correct to use `os.path.join` here? SFTP only supports `/` (I think) and this is platform dependant.

It’s likely more correct to use `posixpath.join` instead.

Comment (ajinkya-xyz) [reply airflow/providers/sftp/sensors/sftp.py]: @uranusjr Makes sense! Updated code to use posixpath instead. Thank you!

Reference : https://stackoverflow.com/questions/36592213/using-os-path-for-posix-path-operations-on-windows

**Comment by uranusjr** [comment]
It seems like the tests need some work to fix temp file usasges.

**Comment by potiuk** [comment]
Yep. Fix would be nice here.

**Comment by ajinkya-xyz** [comment]
@uranusjr @potiuk Facing issue in doing local setup on windows machine. Will try to spend some time on weekend.

**Pull request closed without merging**